[tests] bump node@16 fixtures to node@18

trek opened the pull request:
# PLEASE DO NOT MERGE UNTIL 2025-01-31

Certain fixtures reference node@16 directly in their `engines.node` field. Bump these to 18.

trek committed:
[tests] bump node@16 fixtures to node@18

erikareads reviewed (dismissed):
blocking merge with a request changes

trek committed:
Update some more fixtures

trek committed:
Remove ember exmaple change

trek committed:
Shift buikd utils fixturse

trek committed:
Merge branch 'main' into trek/zero-1753-update-andor-delete-vercelvercel-test-fixtures-that-rely-on

trek committed:
Restore more build-utils fixtures

trek committed:
Merge branch 'main' into trek/zero-1753-update-andor-delete-vercelvercel-test-fixtures-that-rely-on

trek commented:
Shifting this PR to https://github.com/vercel/vercel/pull/12910

trek committed:
Remove all hard-coded node 18 engine entries

TooTallNate committed:
[functions] Publish release after #12758 (#12938)

The changes in #12758 moved code from `@vercel/edge` to
`@vercel/functions`, however the changeset did not reference the
functions package so the moved code was not published in a release.

This PR adds a changeset to trigger a publish of that moved code.

onsclom committed:
[tests] Skip flakey test (#12936)

`vercel/vercel` unit tests on windows on node 22 flakes with module
missing errors. This skips them.

trek committed:
[tests] Remove hard-coded node version from `packages/cli` tests (#12933)

We decided to remove any "engines" value in fixtures that are not
directly related to the test in question. Node 16 is being removed so
starting by removing these since they'll stop working on Friday.

TooTallNate committed:
[cli] Remove a couple logs related e2e tests (#12939)

These tests are flaky and we already have adequate coverage for this
functionality as unit tests.

erikareads committed:
[remix] Create an interface for differences remix vs react-router (#12925)

An attempt at tidying up the logic per:
https://github.com/vercel/vercel/pull/12904#discussion_r1929370988

vercel-release-bot committed:
Version Packages (#12851)

This PR was opened by the [Changesets
release](https://github.com/changesets/action) GitHub action. When
you're ready to do a release, you can merge this and the packages will
be published to npm automatically. If you're not ready to do a release
yet, that's fine, whenever you add more changesets to main, this PR will
be updated.


# Releases
## @vercel/react-router@1.0.0

### Major Changes

- Initial release
([#12903](https://github.com/vercel/vercel/pull/12903))

## @vercel/build-utils@9.1.0

### Minor Changes

- Add `useWebApi` property to `NodejsLambda` class
([#12873](https://github.com/vercel/vercel/pull/12873))

- [build-utils] convert NodeVersion to class and add state getter
([#12883](https://github.com/vercel/vercel/pull/12883))
    [ruby] convert RubyVersion to class and add state getter

## vercel@39.4.0

### Minor Changes

- [cli] add Node Version to project list output table
([#12895](https://github.com/vercel/vercel/pull/12895))

### Patch Changes

- Enable executable bit for `src/vc.js` script
([#12863](https://github.com/vercel/vercel/pull/12863))

- Add video integration category to the integration install CLI
([#12870](https://github.com/vercel/vercel/pull/12870))

- [cli] update warning in ls --update-required for clarity
([#12864](https://github.com/vercel/vercel/pull/12864))

- Added headers for user-supplied rewrites
([#12847](https://github.com/vercel/vercel/pull/12847))

- [cli] refactor confirm component to use `client.input.confirm`
interface ([#12846](https://github.com/vercel/vercel/pull/12846))

- [build-utils] convert NodeVersion to class and add state getter
([#12883](https://github.com/vercel/vercel/pull/12883))
    [ruby] convert RubyVersion to class and add state getter

- Updated dependencies
\[[`56f72ffefb828c3001f7c82259ed0f71db4f629e`](https://github.com/vercel/vercel/commit/56f72ffefb828c3001f7c82259ed0f71db4f629e),
[`745404610a836fa6c2068c5c192d2f3e8b86918f`](https://github.com/vercel/vercel/commit/745404610a836fa6c2068c5c192d2f3e8b86918f),
[`b6bb709370d2565121808340d43d0a9d90b53de1`](https://github.com/vercel/vercel/commit/b6bb709370d2565121808340d43d0a9d90b53de1),
[`85a64db8d9b6a6e9f7c6a019d51777930a806584`](https://github.com/vercel/vercel/commit/85a64db8d9b6a6e9f7c6a019d51777930a806584),
[`ac6a62eff7db3b3e2713adc1f5bd6a1331d838ce`](https://github.com/vercel/vercel/commit/ac6a62eff7db3b3e2713adc1f5bd6a1331d838ce),
[`f65ea2245382dd2449c96f695ed692d419a83868`](https://github.com/vercel/vercel/commit/f65ea2245382dd2449c96f695ed692d419a83868),
[`7af40c1e9d9916cbe8287359662fc940f1d24fce`](https://github.com/vercel/vercel/commit/7af40c1e9d9916cbe8287359662fc940f1d24fce),
[`06307180c20ca07b65f7cb5e93b65c977b9ccd70`](https://github.com/vercel/vercel/commit/06307180c20ca07b65f7cb5e93b65c977b9ccd70),
[`3a5507fd1459c77b4491f1c9c3a64ad42e4ff009`](https://github.com/vercel/vercel/commit/3a5507fd1459c77b4491f1c9c3a64ad42e4ff009),
[`cefda60a603d60cc35e4697c36e751cca411e6bb`](https://github.com/vercel/vercel/commit/cefda60a603d60cc35e4697c36e751cca411e6bb),
[`003866c4c93e893edb7a5e89b519fc8879b370c0`](https://github.com/vercel/vercel/commit/003866c4c93e893edb7a5e89b519fc8879b370c0)]:
    -   @vercel/remix-builder@5.1.0
    -   @vercel/build-utils@9.1.0
    -   @vercel/node@5.0.3
    -   @vercel/python@4.7.1
    -   @vercel/next@4.4.3
    -   @vercel/ruby@2.2.0
    -   @vercel/static-build@2.5.42

## @vercel/frameworks@3.5.0

### Minor Changes

- Make vite detection supersede ionic-react
([#12880](https://github.com/vercel/vercel/pull/12880))

## @vercel/functions@1.6.0

### Minor Changes

- Add middleware-related helper functions
([#12938](https://github.com/vercel/vercel/pull/12938))

## @vercel/remix-builder@5.1.0

### Minor Changes

- Add support for React Router v7
([#12904](https://github.com/vercel/vercel/pull/12904))

- Enable `nativeFetch` when `v3_singleFetch` future flag is enabled
([#12918](https://github.com/vercel/vercel/pull/12918))

### Patch Changes

- [remix] Create an interface for differences remix vs react-router
([#12925](https://github.com/vercel/vercel/pull/12925))

## @vercel/ruby@2.2.0

### Minor Changes

- [build-utils] convert NodeVersion to class and add state getter
([#12883](https://github.com/vercel/vercel/pull/12883))
    [ruby] convert RubyVersion to class and add state getter

## @vercel/client@13.5.5

### Patch Changes

- Updated dependencies
\[[`745404610a836fa6c2068c5c192d2f3e8b86918f`](https://github.com/vercel/vercel/commit/745404610a836fa6c2068c5c192d2f3e8b86918f),
[`3a5507fd1459c77b4491f1c9c3a64ad42e4ff009`](https://github.com/vercel/vercel/commit/3a5507fd1459c77b4491f1c9c3a64ad42e4ff009)]:
    -   @vercel/build-utils@9.1.0

## @vercel/firewall@0.1.6

### Patch Changes

- Bump next from 14.2.10 to 14.2.21
([#12842](https://github.com/vercel/vercel/pull/12842))

## @vercel/fs-detectors@5.3.3

### Patch Changes

- Fix local file system readdir to not throw on special file system stat
types ([#12915](https://github.com/vercel/vercel/pull/12915))

- Updated dependencies
\[[`d645bdd4312730b10bef89ad9e18e111500849fc`](https://github.com/vercel/vercel/commit/d645bdd4312730b10bef89ad9e18e111500849fc)]:
    -   @vercel/frameworks@3.5.0

## @vercel/gatsby-plugin-vercel-builder@2.0.64

### Patch Changes

- Updated dependencies
\[[`745404610a836fa6c2068c5c192d2f3e8b86918f`](https://github.com/vercel/vercel/commit/745404610a836fa6c2068c5c192d2f3e8b86918f),
[`3a5507fd1459c77b4491f1c9c3a64ad42e4ff009`](https://github.com/vercel/vercel/commit/3a5507fd1459c77b4491f1c9c3a64ad42e4ff009)]:
    -   @vercel/build-utils@9.1.0

## @vercel/next@4.4.3

### Patch Changes

- ensure defaultLocale rewrite doesn't conflict with user-defined
redirects ([#12916](https://github.com/vercel/vercel/pull/12916))

- Added headers for user-supplied rewrites
([#12847](https://github.com/vercel/vercel/pull/12847))

## @vercel/node@5.0.3

### Patch Changes

- Fix requests failing due to the presence of `Transfer-Encoding` header
in edge-function dev server.
([#10701](https://github.com/vercel/vercel/pull/10701))

- Split `build()`, `prepareCache()` and `startDevServer()` into separate
files ([#12872](https://github.com/vercel/vercel/pull/12872))

- Updated dependencies
\[[`745404610a836fa6c2068c5c192d2f3e8b86918f`](https://github.com/vercel/vercel/commit/745404610a836fa6c2068c5c192d2f3e8b86918f),
[`3a5507fd1459c77b4491f1c9c3a64ad42e4ff009`](https://github.com/vercel/vercel/commit/3a5507fd1459c77b4491f1c9c3a64ad42e4ff009)]:
    -   @vercel/build-utils@9.1.0

## @vercel/python@4.7.1

### Patch Changes

- Remove support for VERCEL_IPC_FD
([#12908](https://github.com/vercel/vercel/pull/12908))

- Add `supportsResponseStreaming` to build output
([#12884](https://github.com/vercel/vercel/pull/12884))

## @vercel/static-build@2.5.42

### Patch Changes

-   Updated dependencies \[]:
    -   @vercel/gatsby-plugin-vercel-builder@2.0.64

## @vercel-internals/types@3.0.5

### Patch Changes

- Updated dependencies
\[[`745404610a836fa6c2068c5c192d2f3e8b86918f`](https://github.com/vercel/vercel/commit/745404610a836fa6c2068c5c192d2f3e8b86918f),
[`3a5507fd1459c77b4491f1c9c3a64ad42e4ff009`](https://github.com/vercel/vercel/commit/3a5507fd1459c77b4491f1c9c3a64ad42e4ff009)]:
    -   @vercel/build-utils@9.1.0

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

ztanner committed:
[next]: add flag for behavior introduced in #12916 (#12941)

This flags the behavior introduced in:

- #12916

With it by default being disabled to minimize the risk of the change.

vercel-release-bot committed:
Version Packages (#12944)

This PR was opened by the [Changesets
release](https://github.com/changesets/action) GitHub action. When
you're ready to do a release, you can merge this and the packages will
be published to npm automatically. If you're not ready to do a release
yet, that's fine, whenever you add more changesets to main, this PR will
be updated.


# Releases
## vercel@39.4.1

### Patch Changes

- Updated dependencies
\[[`f3d60336f560b6458bc639047efb80c804ea1fbb`](https://github.com/vercel/vercel/commit/f3d60336f560b6458bc639047efb80c804ea1fbb)]:
    -   @vercel/next@4.4.4

## @vercel/next@4.4.4

### Patch Changes

- flag behavior to defer defaultLocale rewrite
([#12941](https://github.com/vercel/vercel/pull/12941))

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

TooTallNate committed:
Remove "examples" changesets (#12945)

The "examples" package is marked as `private: true` in package.json, and
including a changeset for it causes publish to fail.

---------

Co-authored-by: Sean Massa <EndangeredMassa@gmail.com>

TooTallNate committed:
Force publish

TooTallNate committed:
[react-router] Publish as "public" on npm (#12948)

To fix error from
https://github.com/vercel/vercel/actions/runs/13023023388/job/36327301788:

> npm ERR! code EUSAGE
> npm ERR! Can't generate provenance for new or private package, you
must set `access` to public.

__Note:__ Intentionally not adding a changeset here so that changesets
publishes upon merge.

EndangeredMassa committed:
[node][remix][routing-utils] log diff between current and updated versions of `path-to-regexp` (#12926)

Using [pnpm aliases](https://pnpm.io/aliases), this PR installs two
different versions of `path-to-regexp`. Both are run and the resulting
regex is compared. If they are different, an innocuous log message is
sent to the build logs.

This allows us to test out the new version and be notified of any
breaking changes it would have for our users.

---

Testing: This will be manually tested in the build container before
release.

---------

Co-authored-by: Erika Rowland <130414394+erikareads@users.noreply.github.com>

vercel-release-bot committed:
Version Packages (#12952)

This PR was opened by the [Changesets
release](https://github.com/changesets/action) GitHub action. When
you're ready to do a release, you can merge this and the packages will
be published to npm automatically. If you're not ready to do a release
yet, that's fine, whenever you add more changesets to main, this PR will
be updated.


# Releases
## vercel@39.4.2

### Patch Changes

- Updated dependencies
\[[`16b38a92574695f9961c7cc00cf631fe434c26c8`](https://github.com/vercel/vercel/commit/16b38a92574695f9961c7cc00cf631fe434c26c8)]:
    -   @vercel/node@5.0.4
    -   @vercel/remix-builder@5.1.1
    -   @vercel/redwood@2.1.13
    -   @vercel/static-build@2.5.43

## @vercel/client@13.5.6

### Patch Changes

- Updated dependencies
\[[`16b38a92574695f9961c7cc00cf631fe434c26c8`](https://github.com/vercel/vercel/commit/16b38a92574695f9961c7cc00cf631fe434c26c8)]:
    -   @vercel/routing-utils@5.0.1

## @vercel/fs-detectors@5.3.4

### Patch Changes

- Updated dependencies
\[[`16b38a92574695f9961c7cc00cf631fe434c26c8`](https://github.com/vercel/vercel/commit/16b38a92574695f9961c7cc00cf631fe434c26c8)]:
    -   @vercel/routing-utils@5.0.1

## @vercel/gatsby-plugin-vercel-builder@2.0.65

### Patch Changes

- Updated dependencies
\[[`16b38a92574695f9961c7cc00cf631fe434c26c8`](https://github.com/vercel/vercel/commit/16b38a92574695f9961c7cc00cf631fe434c26c8)]:
    -   @vercel/routing-utils@5.0.1

## @vercel/node@5.0.4

### Patch Changes

- log diff between current and updated versions of path-to-regexp
([#12926](https://github.com/vercel/vercel/pull/12926))

## @vercel/redwood@2.1.13

### Patch Changes

- Updated dependencies
\[[`16b38a92574695f9961c7cc00cf631fe434c26c8`](https://github.com/vercel/vercel/commit/16b38a92574695f9961c7cc00cf631fe434c26c8)]:
    -   @vercel/routing-utils@5.0.1

## @vercel/remix-builder@5.1.1

### Patch Changes

- log diff between current and updated versions of path-to-regexp
([#12926](https://github.com/vercel/vercel/pull/12926))

## @vercel/routing-utils@5.0.1

### Patch Changes

- log diff between current and updated versions of path-to-regexp
([#12926](https://github.com/vercel/vercel/pull/12926))

## @vercel/static-build@2.5.43

### Patch Changes

-   Updated dependencies \[]:
    -   @vercel/gatsby-plugin-vercel-builder@2.0.65

## @vercel-internals/types@3.0.6

### Patch Changes

- Updated dependencies
\[[`16b38a92574695f9961c7cc00cf631fe434c26c8`](https://github.com/vercel/vercel/commit/16b38a92574695f9961c7cc00cf631fe434c26c8)]:
    -   @vercel/routing-utils@5.0.1

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

trek committed:
[tests] Remove hard-coded node version from `packages/build-utils` tests (#12940)

We decided to remove any "engines" value in fixtures that are not
directly related to the test in question. Node 16 is being removed so
starting by removing these since they'll stop working on Friday.

There _are_ two fixtures that I believe are intended to test node 16
specifically:

*
[`20-npm-7`](https://github.com/vercel/vercel/blob/7dd88ef2b6dabf231056a181de4604d042c46c49/packages/build-utils/test/fixtures/20-npm-7/package.json)
*
[`15-npm-8-legacy-peer-deps`](https://github.com/vercel/vercel/blob/7dd88ef2b6dabf231056a181de4604d042c46c49/packages/build-utils/test/fixtures/15-npm-8-legacy-peer-deps/package.json)

so these will be removed in a different PR.

Co-authored-by: Sean Massa <EndangeredMassa@gmail.com>

TooTallNate committed:
Ignore "examples" and "@vercel-internals" scoped packages in changesets config (#12946)

Follow-up to #12945 to ensure that it can't happen again.

erikareads committed:
[remix] extract common edge file tracing logic (#12953)

This PR extracts common logic between the node file trace logic for edge
functions in the React Router and Remix createRender functions.

This PR is minimal refactor that extracts the logic, in a future PR it
may be worth creating an "Edge File Trace" function that completely
encapsulates this logic.

ijjk committed:
[next] Add test case for locale false rewrite (#12959)

Just adds a follow-up test case for
https://github.com/vercel/vercel/pull/12916

trek committed:
[static-build] update 45-node-modules-bin-path fixture (#12957)

The existing fixture is quite old, so I've updated it to latest major
and updated the behavior to match how Next 15 accomplishes it.

The old behavior as implemented warned:

> `next export` has been removed in favor of 'output: export' in
next.config.js.

onsclom committed:
[tests] Skip flakey windows test more surgically (#12955)

Instead of skipping all win32 - node 22 - cli unit test files, we can
just skip the one problematic file:
`packages/cli/test/unit/commands/build/index.test.ts`

erikareads committed:
[remix] skip broken test (#12966)

This test is failing due to a known corepack issue, and it's blocking
react router work.

erikareads committed:
[remix] extract common function options (#12960)

This PR assumes that Remix and React-Router should vary at the same
time, and that there isn't any situation where these options should
differ between the two build paths. If that assumption is invalid, we
should reject this PR and consider another approach.

erikareads committed:
[remix] refactor framework slug into FrameworkSettings (#12958)

Now that we have the `FrameworkSettings` interface, it seems like the
correct place for the framework slug to live.

trek committed:
[tests] Update docusaurus-2 fixture (#12970)

Existing version (alpha) has some dependencies that seem to need node
16. Update to support latest node.

erikareads committed:
[remix] extract sourceSearchValue into FrameworkSettings (#12969)

The `searchValue` used for the source seems to be the same across both
edge and node functions, so this PR extracts into the FrameworkSettings.

ijjk committed:
[routing-utils] Update routes schema for new routes limit (#12968)

We updated our default limit to 2048 and have updated documentation so
this schema should be updated as well to match.

onsclom committed:
[tests] fix docusaurus 2 fixture (#12978)

https://github.com/vercel/vercel/pull/12970 incorrectly passed checks
because it failed to deploy, so this fixture is now failing on main.

onsclom committed:
[cli][client] Make split-tgz default for archive deployments (#12943)

This makes `vercel deploy --archive=tgz` use the new `vercel deploy
--archive=split-tgz` behavior.

`vercel deploy --archive=split-tgz` still works, but now it prints a
warning:

<img width="923" alt="image"
src="https://github.com/user-attachments/assets/a650cef7-3917-4435-a379-0f2f240937f6"
/>

erikareads committed:
[remix] remove check based on never set remixRunVercelPkgJson (#12971)

`remixRunVercelPkgJson` was never set, meaning that the condition in
https://github.com/vercel/vercel/pull/12971/files#diff-b96e3bcdb3ec62681992763d9af0f86e19b0c42ba892a0930cc42527a644e99bL650
would always fail. This PR removes the dead code path.

onsclom committed:
[client] Ignore .yarn/cache by default (#12977)

`vercel deploy` currently deploys `.yarn/cache` by default. This adds
`.yarn/cache` to the default [ignored files and
folders](https://vercel.com/docs/deployments/build-features#ignored-files-and-folders)
list, making it ignored by default.

trek committed:
[tests] remove scully-v0 fixture (#12975)

We maintain tests for more recent versions of Scully. The v0 fixture
would need updating to support Node > 16 which is being removed on
2025-01-31. Removing it instead.

trek committed:
[cli] shift node 16 deprecation date to 2025-02-03 (#12981)

Pushing this back a work day just to avoid weekend firedrills.

vercel-release-bot committed:
Version Packages (#12956)

This PR was opened by the [Changesets
release](https://github.com/changesets/action) GitHub action. When
you're ready to do a release, you can merge this and the packages will
be published to npm automatically. If you're not ready to do a release
yet, that's fine, whenever you add more changesets to main, this PR will
be updated.


# Releases
## vercel@40.0.0

### Major Changes

- Make split-tgz the default for archive deployments
([#12943](https://github.com/vercel/vercel/pull/12943))

### Patch Changes

- Updated dependencies
\[[`a9f2138035befec646d01c23203dba66790ea323`](https://github.com/vercel/vercel/commit/a9f2138035befec646d01c23203dba66790ea323),
[`e9acda308cefc4e12784d742968a6816e1f1072b`](https://github.com/vercel/vercel/commit/e9acda308cefc4e12784d742968a6816e1f1072b),
[`bfbe07e2b5a87b07da464a6504566f29a76a9e8b`](https://github.com/vercel/vercel/commit/bfbe07e2b5a87b07da464a6504566f29a76a9e8b),
[`d07466a32f9026109f333fe653d1ff02a012756f`](https://github.com/vercel/vercel/commit/d07466a32f9026109f333fe653d1ff02a012756f),
[`96c7a3ab6041efeb1eda9aab97d0a75c61e17bbe`](https://github.com/vercel/vercel/commit/96c7a3ab6041efeb1eda9aab97d0a75c61e17bbe),
[`b52b7e3a8cd775d56149683cb809b7ad9c77a514`](https://github.com/vercel/vercel/commit/b52b7e3a8cd775d56149683cb809b7ad9c77a514),
[`9cfe0ec1439c6d3b038ab21f125cfe0223ad62c8`](https://github.com/vercel/vercel/commit/9cfe0ec1439c6d3b038ab21f125cfe0223ad62c8)]:
    -   @vercel/remix-builder@5.1.2
    -   @vercel/next@4.4.5
    -   @vercel/build-utils@9.1.1
    -   @vercel/node@5.0.5
    -   @vercel/redwood@2.1.14
    -   @vercel/static-build@2.5.44

## @vercel/client@14.0.0

### Major Changes

- Ignore .yarn/cache by default on vercel CLI deploys
([#12977](https://github.com/vercel/vercel/pull/12977))

- Make split-tgz the default for archive deployments
([#12943](https://github.com/vercel/vercel/pull/12943))

### Patch Changes

- Updated dependencies
\[[`b52b7e3a8cd775d56149683cb809b7ad9c77a514`](https://github.com/vercel/vercel/commit/b52b7e3a8cd775d56149683cb809b7ad9c77a514),
[`3d310f508b9ff8f891838d8ba8ea24e428a06a9d`](https://github.com/vercel/vercel/commit/3d310f508b9ff8f891838d8ba8ea24e428a06a9d)]:
    -   @vercel/build-utils@9.1.1
    -   @vercel/routing-utils@5.0.2

## @vercel/build-utils@9.1.1

### Patch Changes

- [cli] shift node 16 deprecation date to 2025-02-03
([#12981](https://github.com/vercel/vercel/pull/12981))

## @vercel/fs-detectors@5.3.5

### Patch Changes

- Updated dependencies
\[[`3d310f508b9ff8f891838d8ba8ea24e428a06a9d`](https://github.com/vercel/vercel/commit/3d310f508b9ff8f891838d8ba8ea24e428a06a9d)]:
    -   @vercel/routing-utils@5.0.2

## @vercel/gatsby-plugin-vercel-builder@2.0.66

### Patch Changes

- Updated dependencies
\[[`b52b7e3a8cd775d56149683cb809b7ad9c77a514`](https://github.com/vercel/vercel/commit/b52b7e3a8cd775d56149683cb809b7ad9c77a514),
[`3d310f508b9ff8f891838d8ba8ea24e428a06a9d`](https://github.com/vercel/vercel/commit/3d310f508b9ff8f891838d8ba8ea24e428a06a9d)]:
    -   @vercel/build-utils@9.1.1
    -   @vercel/routing-utils@5.0.2

## @vercel/next@4.4.5

### Patch Changes

- Add test case for locale false rewrite
([#12959](https://github.com/vercel/vercel/pull/12959))

## @vercel/node@5.0.5

### Patch Changes

- Updated dependencies
\[[`b52b7e3a8cd775d56149683cb809b7ad9c77a514`](https://github.com/vercel/vercel/commit/b52b7e3a8cd775d56149683cb809b7ad9c77a514)]:
    -   @vercel/build-utils@9.1.1

## @vercel/redwood@2.1.14

### Patch Changes

- Updated dependencies
\[[`3d310f508b9ff8f891838d8ba8ea24e428a06a9d`](https://github.com/vercel/vercel/commit/3d310f508b9ff8f891838d8ba8ea24e428a06a9d)]:
    -   @vercel/routing-utils@5.0.2

## @vercel/remix-builder@5.1.2

### Patch Changes

- [remix] extract common function options
([#12960](https://github.com/vercel/vercel/pull/12960))

- [remix] extract sourceSearchValue into FrameworkSettings
([#12969](https://github.com/vercel/vercel/pull/12969))

- [remix] extract common edge file tracing logic
([#12953](https://github.com/vercel/vercel/pull/12953))

- [remix] remove check based on never set remixRunVercelPkgJson
([#12971](https://github.com/vercel/vercel/pull/12971))

- [remix] refactor framework slug into FrameworkSettings
([#12958](https://github.com/vercel/vercel/pull/12958))

## @vercel/routing-utils@5.0.2

### Patch Changes

- Update routes schema for new limit of 2048
([#12968](https://github.com/vercel/vercel/pull/12968))

## @vercel/static-build@2.5.44

### Patch Changes

-   Updated dependencies \[]:
    -   @vercel/gatsby-plugin-vercel-builder@2.0.66

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

TooTallNate committed:
[react-router] Ensure proper files are published (#12984)

A Turbo cache hit during publish caused the build command to be skipped
during publish, and thus the compiled `*.js` and `*.d.ts` files are
missing in the `@vercel/react-router@1.0.0` release.

vercel-release-bot committed:
Version Packages (#12985)

This PR was opened by the [Changesets
release](https://github.com/changesets/action) GitHub action. When
you're ready to do a release, you can merge this and the packages will
be published to npm automatically. If you're not ready to do a release
yet, that's fine, whenever you add more changesets to main, this PR will
be updated.


# Releases
## @vercel/react-router@1.0.1

### Patch Changes

- Ensure proper files are published
([#12984](https://github.com/vercel/vercel/pull/12984))

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

TooTallNate committed:
[react-router] Fix "entry.server" Edge runtime condition (#12986)

The proper name is "edge-light", not "edge".

vercel-release-bot committed:
Version Packages (#12987)

This PR was opened by the [Changesets
release](https://github.com/changesets/action) GitHub action. When
you're ready to do a release, you can merge this and the packages will
be published to npm automatically. If you're not ready to do a release
yet, that's fine, whenever you add more changesets to main, this PR will
be updated.


# Releases
## @vercel/react-router@1.0.2

### Patch Changes

- Fix "entry.server" Edge runtime condition
([#12986](https://github.com/vercel/vercel/pull/12986))

TooTallNate committed:
[frameworks] Add "react-router" framework entry (#12932)

Adds a new framework entry for React Router (when being used as a
framework). The same Remix builder will be used internally, but having a
dedicated preset allows for auto-detection to work and will show the
correct logo when being used in the Project Settings.

(**Note:** this is a re-application of #12905, which was reverted in
#12927)

onsclom committed:
Add yarn cache to default build cache paths (#12961)

Adds `.yarn/cache` to the `prepareCache` for builders that may be using
yarn. This allows yarn 2, 3, and 4 projects to cache deps between
deploys.

Tested manually with each major yarn version using `vercel deploy -b
VERCEL_CLI_VERSION=https://vercel-i1pucqxc7.vercel.sh/tarballs/vercel.tgz`
and confirmed this cached deps on each version.

TooTallNate committed:
Include CLI release as part of new "React Router" framework preset (#12989)

Since `@vercel/frameworks` is a dev dependency in CLI, changesets
doesn't qualify CLI for a release when the dependency is released.

So include CLI in the changeset to forcefully include it in the release.

vercel-release-bot committed:
Version Packages (#12988)

This PR was opened by the [Changesets
release](https://github.com/changesets/action) GitHub action. When
you're ready to do a release, you can merge this and the packages will
be published to npm automatically. If you're not ready to do a release
yet, that's fine, whenever you add more changesets to main, this PR will
be updated.


# Releases
## @vercel/build-utils@9.2.0

### Minor Changes

- Add .yarn/cache to build cache
([#12961](https://github.com/vercel/vercel/pull/12961))

## vercel@40.1.0

### Minor Changes

- Add "react-router" framework entry
([#12932](https://github.com/vercel/vercel/pull/12932))

### Patch Changes

- Updated dependencies
\[[`5c696af2b40f0fc368e84cafa6d82b2ce998fc19`](https://github.com/vercel/vercel/commit/5c696af2b40f0fc368e84cafa6d82b2ce998fc19)]:
    -   @vercel/static-build@2.6.0
    -   @vercel/build-utils@9.2.0
    -   @vercel/hydrogen@1.1.0
    -   @vercel/redwood@2.2.0
    -   @vercel/remix-builder@5.2.0
    -   @vercel/next@4.5.0
    -   @vercel/node@5.1.0

## @vercel/frameworks@3.6.0

### Minor Changes

- Add "react-router" framework entry
([#12932](https://github.com/vercel/vercel/pull/12932))

## @vercel/hydrogen@1.1.0

### Minor Changes

- Add .yarn/cache to build cache
([#12961](https://github.com/vercel/vercel/pull/12961))

## @vercel/next@4.5.0

### Minor Changes

- Add .yarn/cache to build cache
([#12961](https://github.com/vercel/vercel/pull/12961))

## @vercel/node@5.1.0

### Minor Changes

- Add .yarn/cache to build cache
([#12961](https://github.com/vercel/vercel/pull/12961))

### Patch Changes

- Updated dependencies
\[[`5c696af2b40f0fc368e84cafa6d82b2ce998fc19`](https://github.com/vercel/vercel/commit/5c696af2b40f0fc368e84cafa6d82b2ce998fc19)]:
    -   @vercel/build-utils@9.2.0

## @vercel/redwood@2.2.0

### Minor Changes

- Add .yarn/cache to build cache
([#12961](https://github.com/vercel/vercel/pull/12961))

## @vercel/remix-builder@5.2.0

### Minor Changes

- Add .yarn/cache to build cache
([#12961](https://github.com/vercel/vercel/pull/12961))

## @vercel/static-build@2.6.0

### Minor Changes

- Add .yarn/cache to build cache
([#12961](https://github.com/vercel/vercel/pull/12961))

### Patch Changes

-   Updated dependencies \[]:
    -   @vercel/gatsby-plugin-vercel-builder@2.0.67

## @vercel/client@14.0.1

### Patch Changes

- Updated dependencies
\[[`5c696af2b40f0fc368e84cafa6d82b2ce998fc19`](https://github.com/vercel/vercel/commit/5c696af2b40f0fc368e84cafa6d82b2ce998fc19)]:
    -   @vercel/build-utils@9.2.0

## @vercel/fs-detectors@5.3.6

### Patch Changes

- Updated dependencies
\[[`f9d8407866ac95db70dfd961cc51a6d2df233efa`](https://github.com/vercel/vercel/commit/f9d8407866ac95db70dfd961cc51a6d2df233efa)]:
    -   @vercel/frameworks@3.6.0

## @vercel/gatsby-plugin-vercel-builder@2.0.67

### Patch Changes

- Updated dependencies
\[[`5c696af2b40f0fc368e84cafa6d82b2ce998fc19`](https://github.com/vercel/vercel/commit/5c696af2b40f0fc368e84cafa6d82b2ce998fc19)]:
    -   @vercel/build-utils@9.2.0

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

erikareads committed:
[remix] extract common handler logic into function (#12973)

The logic around `handler` and `handlerPath` was consistent across all
three ServerFunction creation functions, so this PR extracts that logic
into a common function so that they stay in sync with each other.

erikareads committed:
[remix] extract common logic to get files from node file trace (#12983)

After #12971 the logic for getting files from node file trace is common
across the ServerFunction creation functions. This PR extracts that
logic into a function.

trek committed:
[tests] remove saber fixture (#12972)

Follow up to https://github.com/vercel/vercel/pull/12923

---------

Co-authored-by: Erika Rowland <130414394+erikareads@users.noreply.github.com>

trek committed:
Remove gridsome example and fixtures (#12974)

Gridsome's last release was `0.7.23` on 2020-11-22. It would require
some work to update to work in node > 16, which we are removing support
for on 2025-01-31. Opting to remove the example and fixture instead.

trek committed:
Merge branch 'trek/zero-1753-update-andor-delete-vercelvercel-test-fixtures-that-rely-on' of github.com:vercel/vercel into trek/zero-1753-update-andor-delete-vercelvercel-test-fixtures-that-rely-on

trek commented:
Closing. We decided to entirely remove the engines entry entirely
